staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -260,9 +263,15 @@ func (c *Config) ApplySecureServingOptions(secureServing *options.SecureServingO
 		return c, nil
 	}
 
+	ipMax := c.MaxRequestsInFlight / 2
+	if ipMax > c.MaxMutatingRequestsInFlight/2 {
+		ipMax = c.MaxMutatingRequestsInFlight / 2
+	}
+	max := c.MaxRequestsInFlight + c.MaxMutatingRequestsInFlight
```

Comment:
The only long running feature I am aware of is --long-running-request-regexp "which should be excluded from maximum inflight request handling". So that bucket does not have a pool size per se. It seemed simplest to me  to claim this was not part of "max inflight request handling". In part I was advocating for this as at the time we are enforcing the ipLimit we don't know what the URL will be for the regexp match. If we decide to do it, we will probably need to fix the regexp code to call the decrementer on a match.